process/block/poolsCleaner/txsPoolsCleaner.go

```diff
@@ -178,10 +178,11 @@ func (tpc *txsPoolsCleaner) processReceivedTx(
 	receiverShardID uint32,
 	txType int8,
 ) {
-	tpc.mutMapTxsRounds.Lock()
-	defer tpc.mutMapTxsRounds.Unlock()
+	tpc.mutMapTxsRounds.RLock()
```

Comment:
Now I see that the cleaner removes each tx individually, it doesn't call the bulk operations (e.g. `RemoveSetOfDataFromPool`) but that should be fine nonetheless.

Reply:
Yes, because we don't have such a method implemented, but anyway the cleanTxsPoolsIfNeeded method is ran only once per minute.